app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx

```diff
@@ -664,6 +687,8 @@ class DatasourceRestAPIEditor extends React.Component<Props> {
             ]}
           />
         </FormInputContainer>
+        {!_.get(formData.authentication, "isAuthorizationHeader") &&
```

Comment:
Set a default value so you will always get false if this field does not exist for some reason
```suggestion
        {!_.get(formData.authentication, "isAuthorizationHeader", false) &&
```

Reply:
I'm defaulting `isAuthorizationHeader` to true when the incoming data is undefined. Should this still be required? @hetunandu